apps/demo-typegen/src/demo.test.ts

```diff
@@ -79,3 +116,46 @@ it('should throw when dry running via contract factory with wallet with no resou
 
   expect((<Error>error).message).toMatch('not enough coins to fit the target');
 });
+
+test('Example script', async () => {
+  const provider = await Provider.create(FUEL_NETWORK_URL);
+  const wallet = await generateTestWallet(provider, [[500_000, BaseAssetId]]);
+
+  // #region typegen-demo-script
+  // #context import { ScriptAbi__factory } from './types';
+
+  const script = ScriptAbi__factory.createInstance(wallet);
+  const { value } = await script.functions
+    .main()
+    .txParams({
+      gasPrice: provider.getGasConfig().minGasPrice,
+    })
+    .call();
+  // #endregion typegen-demo-script
+  // @ts-expect-error TODO: investitage - typegen is expecting value to be a number but the value being returned is the string '0xa'
+  expect(value.toNumber()).toBe(10);
+});
+
+// TODO: investigate - this test is currently failing due to some gas errors. 'not enough coins to fit the target'. even though everything is setup correctly.
+test.skip('Example predicate', async () => {
+  // #region typegen-demo-predicate
+  // #context import { PredicateAbi__factory } from './types';
+  const provider = await Provider.create(FUEL_NETWORK_URL);
+  const wallet = await generateTestWallet(provider, [[500, BaseAssetId]]);
+  const receiver = Wallet.fromAddress(Address.fromRandom(), provider);
+
```

Comment:
It would be great to get this test to pass before merging it.

Cc'ing @FuelLabs/sdk-ts — ideas?

Reply:
This is happening because `gasLimit` is not being set in the transfer methods. Currently, our transfer methods set `gasLimit` to `maxGasPerTx` when it is not being set. 

This reflects in transaction fee validations to be way higher than they will actually cost.

try: 

```ts
  const provider = await Provider.create(FUEL_NETWORK_URL);
  const wallet = await generateTestWallet(provider, [[500, BaseAssetId]]);
  const receiver = Wallet.fromAddress(Address.fromRandom(), provider);

  const predicate = PredicateAbi__factory.createInstance(provider);

  const tx = await wallet.transfer(predicate.address, 150, BaseAssetId, {
    gasPrice: provider.getGasConfig().minGasPrice,
    gasLimit: 50,
  });
  const { fee } = await tx.wait();

  const tx2 = await predicate.transfer(receiver.address, 100, BaseAssetId, {
    gasPrice: provider.getGasConfig().minGasPrice,
    gasLimit: 50,
  });
  await tx2.wait();

  expect((await wallet.getBalance()).toNumber()).toEqual(500 - 150 - fee.toNumber());
  ```

Reply:
Thanks @Torres-ssf, this fixed it. But I am a bit concerned as to why do we have to specify this? I feel like our defaults should be smart enough to avoid such scenarios. What do you think?

Reply:
cc @Torres-ssf